api.js

```diff
@@ -802,19 +813,47 @@ const createTimetable = function(obj) {
             currDate.setDate(currDate.getDate()+1);
           }
 
-          timetable[currDate.toISOString()] = 0;
+          var endDateTime = new Date(currDate.getTime());
+          //increment in half hour intervals
+          if(currMin == 30) {
+            endDateTime.setHours(currHour + 1);
+            endDateTime.setMinutes(0);
+          }
+          else {
+            endDateTime.setMinutes(30);
+          }
+
+          var otherRoomNames = [];
+          for(var t = 0; t < otherRooms.length; t++) {
+            otherRoomNames.push(otherRooms[t].name);
+          }
+
+          console.log(otherRoomNames);
+
+          var timeslot = {
+            startDateTime: currDate.toISOString(),
+            endDateTime: endDateTime,
+            rooms: otherRoomNames,
+            available: 0
+          };
+          //timetable.push(timeslot);
 
           var x = 0;
           //iterate through all meetings. if we find a meeting that conflicts, set that to busy
           //if we don't, then the time is open/free
           while(x < meetings.length) {
             //if time is between a meeting, set timetable[time] to 1 (busy)
-            if(((new Date(meetings[x].start_datetime)).getTime() <= (new Date(currDate.toISOString())).getTime()) && 
-               ((new Date(meetings[x].end_datetime)).getTime() > (new Date(currDate.toISOString())).getTime())) {
-              timetable[currDate.toISOString()] = 1;
+            //TODO: check inequality signs/logic
+            if((((new Date(meetings[x].start_datetime)).getTime() <= (new Date(currDate.toISOString())).getTime()) && 
+                ((new Date(meetings[x].end_datetime)).getTime() > (new Date(currDate.toISOString())).getTime())) ||
+               (((new Date(meetings[x].start_datetime)).getTime() <= (new Date(endDateTime.toISOString())).getTime()) && 
+                ((new Date(meetings[x].end_datetime)).getTime() >= (new Date(endDateTime.toISOString())).getTime()))) {
+              //timetable[timetable.length - 1].available = 1;
+                timeslot.available = 1;
```

Comment:
As per the comment below, I think you should simplify this code by using variable assignment prior to the conditional.

```
const meetingStartTime = ((new Date(meetings[x].start_datetime)).getTime();
```